datafusion/physical-plan/src/common.rs

```diff
@@ -204,12 +204,24 @@ impl<R: 'static> SpawnedTask<R> {
         Self { inner }
     }
 
+    /// Joins the task, returning the result of join (`Result<R, JoinError>`).
     pub async fn join(mut self) -> Result<R, JoinError> {
         self.inner
             .join_next()
             .await
             .expect("`SpawnedTask` instance always contains exactly 1 task")
     }
+
+    /// Joins the task and unwinds the panic if it happens.
+    pub async fn join_unwind(self) -> R {
```

Comment:
Alternatively, we can always unwind in `join`. But for specific cases have `raw_join` or something like that

Reply:
I think the existing naming makes sense to me i.e. call `join_unwind` if you don't want to manually handle the join error.